src/lib/proof-system/zkprogram.ts

```diff
@@ -391,10 +398,30 @@ function ZkProgram<
             `Try calling \`await program.compile()\` first, this will cache provers in the background.\nIf you compiled your zkProgram with proofs disabled (\`proofsEnabled = false\`), you have to compile it with proofs enabled first.`
         );
       }
-      let publicInputFields = toFieldConsts(publicInputType, publicInput);
+
+      let nonPureInputExists = publicInputType
+        .toAuxiliary(publicInput)
+        .some((aux) => aux.length !== 0);
+
+      let publicInputFields, publicInputAux;
+      if (nonPureInputExists) {
```

Comment:
what about unifying both branches here and only using one function `toFieldAndAuxConsts` instead of two?